packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts

```diff
@@ -0,0 +1,213 @@
+import dayjs from 'dayjs';
+import {
+	ButtonInteraction,
+	ButtonStyle,
+	ChatInputCommandInteraction,
+	Collection,
+	ComponentType,
+	GuildMember,
+	ModalSubmitInteraction,
+	Snowflake,
+} from 'discord.js';
+import i18next from 'i18next';
+import { nanoid } from 'nanoid';
+import { DATE_FORMAT_LOGFILE } from '../../../../Constants.js';
+import { blastOff } from '../../../../functions/anti-raid/blastOff.js';
+import { formatMemberTimestamps } from '../../../../functions/anti-raid/formatMemberTimestamps.js';
+import {
+	formatAntiRaidResultsToAttachment,
+	formatMembersToAttachment,
+} from '../../../../functions/logging/formatMembersToAttachment.js';
+import { insertAntiRaidNukeCaseLog } from '../../../../functions/logging/insertAntiRaidNukeCaseLog.js';
+import { upsertAntiRaidNukeReport } from '../../../../functions/logging/upsertGeneralLog.js';
+import { logger } from '../../../../logger.js';
+import { createButton } from '../../../../util/button.js';
+import { createMessageActionRow } from '../../../../util/messageActionRow.js';
+
+export interface IdValidationResult {
+	validMembers: Collection<string, GuildMember>;
+	validIdCount: number;
+	invalidIdCount: number;
+	totalIdCount: number;
+}
+
+export async function validateMemberIds(
+	interaction: ChatInputCommandInteraction<'cached'> | ModalSubmitInteraction<'cached'>,
+	ids: Set<Snowflake>,
+	emptyResultErrorText: string,
+): Promise<IdValidationResult> {
+	const fetchedMembers = await interaction.guild.members.fetch({ force: true });
+	const result = new Collection<string, GuildMember>();
+
+	for (const id of ids) {
+		const member = fetchedMembers.get(id);
+
+		if (member) {
+			result.set(id, member);
+		}
+	}
+
+	if (!result.size) {
+		throw new Error(emptyResultErrorText);
+	}
+
+	return {
+		validMembers: result,
+		validIdCount: result.size,
+		invalidIdCount: ids.size - result.size,
+		totalIdCount: ids.size,
+	};
+}
+
+export async function handleAntiRaidNuke(
+	interaction: ChatInputCommandInteraction<'cached'> | ModalSubmitInteraction<'cached'>,
+	members: Collection<Snowflake, GuildMember>,
+	locale: string,
+	parameterStrings: string[],
+	reason?: string,
+	days?: number,
+) {
+	if (!members.size) {
+		await interaction.editReply({
+			content: `${i18next.t('command.mod.anti_raid_nuke.common.errors.no_hits', {
+				lng: locale,
+			})}\n\n${parameterStrings.join('\n')}`,
+		});
+	}
+
+	const pruneDays = Math.min(Math.max(Number(days ?? 1), 0), 7);
+
+	parameterStrings.splice(
+		0,
+		0,
+		i18next.t('command.mod.anti_raid_nuke.common.parameters.heading', { lng: locale }),
+		i18next.t('command.mod.anti_raid_nuke.common.parameters.days', {
+			count: pruneDays,
+			lng: locale,
+		}),
+	);
```

Comment:
Why mutate like that 🤔

Reply:
looks absolutely horrible, now that i look at it again